tests/mir-opt/instsimplify/ub_check.rs

```diff
@@ -1,4 +1,5 @@
 //@ unit-test: InstSimplify
+//@ should-fail Broken due to https://github.com/rust-lang/rust/issues/124150.
```

Comment:
Why does this affect this test, when https://github.com/rust-lang/rust/pull/120614 did not change this test?

Reply:
This was already accomplished by the existing `SimplifyToIf`. I just did a refactor in #120614.

Reply:
Oh, so this is not equivalent to reverting https://github.com/rust-lang/rust/pull/120614, it disables more than that.

In that case, hard for me to say whether it's appropriate to disable the entire pass.